src/peeker/src/main.rs

```diff
@@ -62,16 +71,23 @@ fn main() -> Result<()> {
     } else {
         let _ = init_result;
         measure_peek_times(&config);
-        unreachable!()
+        Ok(())
     }
 }
 
 fn measure_peek_times(config: &Args) {
     let mut peek_threads = vec![];
-    for group in &config.config.groups {
+    let run_dur = chrono::Duration::seconds(if config.run_secs == 0 {
```

Comment:
Nit: this defaulting should happen in the config struct not down here in the implementation (line 122 above)